test/bilinearinterpolation.jl

```diff
@@ -0,0 +1,209 @@
+            trial_interpolator =
+                @benchmark Metal.@sync interpolatebilinear!($ftarget, $bilinear, $fsource)
+        else
+            trial_interpolator =
+                @benchmark interpolatebilinear!($ftarget, $bilinear, $fsource)
+        end
+    end
+    return (trial_constructor, trial_interpolator)
+end
+#=
+get_dims_singlelevel(::Type{FT}) where {FT} =
+    ((FT(0), FT(3π)), (FT(0), FT(2π)), (200, 150), (300, 400), FT(0.005))
+get_dims_multilevel(::Type{FT}) where {FT} = (
+    (FT(0), FT(3π)),
+    (FT(0), FT(2π)),
+    (FT(0), FT(1)),
+    (200, 150),
+    (300, 400),
+    2048,
+    FT(0.005),
+)
+=#
+get_dims_singlelevel(::Type{FT}) where {FT} =
+    ((FT(0), FT(3π)), (FT(0), FT(2π)), (2560, 1280), (2400, 1200), FT(0.00005))
+get_dims_multilevel(::Type{FT}) where {FT} = (
+    (FT(0), FT(3π)),
+    (FT(0), FT(2π)),
+    (FT(0), FT(1)),
+    (2560, 1280),
+    (2400, 1200),
+    128,
+    FT(0.00005),
+)
+@show "--------------CPU benchmarks----------------"
+xrange, yrange, nsource, ntarget, toler = get_dims_singlelevel(Float32)
+(trial_sl_cpu_ft32_cons, trial_sl_cpu_ft32_interp) =
+    test_single_level(Array, Float32, xrange, yrange, nsource, ntarget, toler = toler)
+println(
+    "interpolation time = $(Statistics.median(trial_sl_cpu_ft32_interp)) on CPU for single level with Float32\nConstructor time = $(Statistics.median(trial_sl_cpu_ft32_cons))",
+)
+@show "--------------------------------------------"
+
+xrange, yrange, nsource, ntarget, toler = get_dims_singlelevel(Float64)
+trial_sl_cpu_ft64_cons, trial_sl_cpu_ft64_interp =
+    test_single_level(Array, Float64, xrange, yrange, nsource, ntarget, toler = toler)
+println(
+    "interpolation time = $(Statistics.median(trial_sl_cpu_ft64_interp)) on CPU for single level with Float64\nConstructor time = $(Statistics.median(trial_sl_cpu_ft64_cons))",
+)
+@show "--------------------------------------------"
+xrange, yrange, zrange, nsource, ntarget, nlevels, toler = get_dims_multilevel(Float32)
+trial_ml_cpu_ft32_cons, trial_ml_cpu_ft32_interp = test_multilevel(
+    Array,
+    Float32,
+    xrange,
+    yrange,
+    zrange,
+    nsource,
+    ntarget,
+    nlevels,
+    toler = toler,
+)
+println(
+    "interpolation time = $(Statistics.median(trial_ml_cpu_ft32_interp)) on CPU for $nlevels levels with Float32\nConstructor time = $(Statistics.median(trial_ml_cpu_ft32_cons))",
+)
+@show "--------------------------------------------"
+xrange, yrange, zrange, nsource, ntarget, nlevels, toler = get_dims_multilevel(Float64)
+trial_ml_cpu_ft64_cons, trial_ml_cpu_ft64_interp = test_multilevel(
+    Array,
+    Float64,
+    xrange,
+    yrange,
+    zrange,
+    nsource,
+    ntarget,
+    nlevels,
+    toler = toler,
+)
+println(
+    "interpolation time = $(Statistics.median(trial_ml_cpu_ft64_interp)) on CPU for $nlevels levels with Float64\nConstructor time = $(Statistics.median(trial_ml_cpu_ft64_cons))",
+)
+@show "--------------Metal benchmarks---------------"
```

Comment:
This should throw exceptions on incompatible devices right ? Could we reorganise the tests by available device extensions?

Reply:
I am still working on cleaning up the tests. I will fix this shortly.